test/exercism/user_test.rb

```diff
@@ -147,6 +147,25 @@ def test_user_daily_count_returns_0_if_no_daily
     assert_equal 0, fred.daily_count
   end
 
+  def test_user_share_key
+    fred = User.create(username: 'fred')
+    assert_equal nil, fred.share_key
+    fred.set_share_key
+    refute_equal nil, fred.share_key
+    fred.unset_share_key
+    assert_equal nil, fred.share_key
+  end
+
+  def test_user_find_by_username_and_share_key
+    username = 'fred'
+    key = Exercism.uuid
+    fred = User.create(username: username)
+    assert_equal nil, User.find_by_username_and_share_key(username, key)
+    fred.set_share_key
+    key = fred.share_key
+    assert_equal username, User.find_by_username_and_share_key(username, key).username
```

Comment:
Go ahead and inline some of these temporary variables when they're only used once.